src/main/java/io/debezium/connector/yugabytedb/YugabyteDBStreamingChangeEventSource.java

```diff
@@ -751,8 +756,8 @@ private void addTabletIfNotPresent(List<Pair<String,String>> tabletPairList,
 
             LOGGER.info("Initialized offset context for tablet {} with OpId {}", tabletId, OpId.from(pair.getCdcSdkCheckpoint()));
 
-            // Add the flag to indicate that we do not need the schema from this tablet again.
-            schemaNeeded.put(tabletId, Boolean.FALSE);
+            // Add the flag to indicate that we need the schema for the new tablets so that the schema can be registered.
+            schemaNeeded.put(tabletId, Boolean.TRUE);
```

Comment:
have we checked that we don't fetch schema everytime?

Reply:
Yes, as soon as we get a DDL, we set this flag to false while processing the records in the for loop.